### PR TITLE
Add pod spec validation

### PR DESCRIPTION
**Pull request by katrogan**
Signed-off-by: Katrina Rogan <katroganGH@gmail.com>

# TL;DR
Now that pod specs are first class task template targets we should add some preliminary validation during compilation since we can reason about the expected object structure.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
fixes https://github.com/flyteorg/flyte/issues/998

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by katrogan** [comment]
ping @EngHabu

**Comment by kumare3** [comment]
This is better. Did you remove this from flyteadmin then?

**Comment by katrogan** [comment]
@kumare3 yep, closed that PR in favor of this after @EngHabu pointed it out

File: go.sum [hunk 1]
```diff
@@ -237,6 +237,8 @@ github.com/flyteorg/flyteplugins v0.5.56/go.mod h1:Jp5WheQMI08luZmgcmcgyjtzakKH0
 github.com/flyteorg/flytestdlib v0.3.13/go.mod h1:Tz8JCECAbX6VWGwFT6cmEQ+RJpZ/6L9pswu3fzWs220=
 github.com/flyteorg/flytestdlib v0.3.17 h1:7OexDLAjTBzJNGMmKKFmUTkss0I9IFo1LdTMpvH4qqA=
 github.com/flyteorg/flytestdlib v0.3.17/go.mod h1:VlbQuHTE+z2N5qusfwi+6WEkeJoqr8Q0E4NtBAsdwkU=
+github.com/flyteorg/flytestdlib v0.3.27 h1:d3OI5qb5u8CkSs2HMTuM62K5GuTrf6FJKq8CHW6Ymbs=
```
[the remark this answers is not in the document]
Comment (katrogan) [reply go.sum:238]: done

**Review by EngHabu — APPROVED** [review]
